test/performance/search/shape_iterator_benchmark.cpp

```diff
@@ -0,0 +1,82 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <random>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/test/performance/units.hpp>
+#include <seqan3/search/kmer_index/shape.hpp>
+#include <seqan3/search/kmer_index/shape_iterator.hpp>
+
+using namespace seqan3;
+
+
+// ===========================================================================
+//  taken from affine_alignment_benchmark.cpp
+template <typename alphabet_t>
+auto generate_sequence_seqan3(size_t const len = 500,
+                              size_t const variance = 0,
+                              size_t const seed = 0)
+{
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<uint8_t> dis_alpha(0, alphabet_size_v<alphabet_t> - 1);
+    std::uniform_int_distribution<size_t> dis_length(len - variance, len + variance);
+
+    std::vector<alphabet_t> sequence;
+
+    size_t length = dis_length(gen);
+    for (size_t l = 0; l < length; ++l)
+        sequence.push_back(alphabet_t{}.assign_rank(dis_alpha(gen)));
+
+    return sequence;
+}
+//  taken from affine_alignment_benchmark.cpp end
+// ============================================================================
+
+static void shape_empty_construct_benchmark(benchmark::State& state)
+{
+    for (auto _ : state)
+        shape s();
+}
+
+static void shape_construct_benchmark(benchmark::State& state)
+{
+    size_t size = state.range(0);
+    for (auto _ : state)
+        shape s(size);
+}
+
+// performance tests auf dna4 ausreichend?
+static void shape_iterator_construct(benchmark::State& state)
```

Comment:
We also don't really need to benchmark this

Reply:
Despite the first hash value is calculated upon construction?

Reply:
Well it's only one hash value. The typical usecase will probably be iterating over a sequence and generating hashes.

Reply:
You can leave this one in if you want, though.